src/s3/commands/downloadFileAs.ts

```diff
@@ -163,3 +169,150 @@ async function promptForSaveLocation(
         filters: filters,
     })
 }
+
+async function downloadBatchFiles(fileList: S3File[], saveLocation: vscode.Uri, client: S3Client): Promise<S3File[]> {
+    const results = await Promise.all(
+        fileList.map(async file => {
+            try {
+                await downloadFile(file, {
+                    window: Window.vscode(),
+                    saveLocation: vscode.Uri.file(path.join(saveLocation.fsPath, file.name)),
+                    client,
+                    progressLocation: vscode.ProgressLocation.Notification,
+                })
+            } catch (error) {
+                getLogger().error(`s3: Failed to download file '${file.name}': `, error as Error)
+                return file
+            }
+        })
+    )
+    return results.filter(isNonNullable)
+}
+
+/**
+ * Downloads an S3 Folder or multiple selected S3 files
+ *
+ * @param node S3FolderNode to download
+ * @param allNodes Multi-selected explorer nodes
+ * @param window
+ * @param outputChannel
+ */
+export async function downloadFolderCommand(
+    node: S3FolderNode | S3FileNode,
+    allNodes: S3FileNode[] = [],
+    window = Window.vscode(),
+    outputChannel = globals.outputChannel
+): Promise<void> {
+    let files: S3File[]
+    let saveLocation = await promptForSaveFolderLocation()
+    if (!saveLocation) {
+        throw new CancellationError('user')
+    }
+
+    if (node instanceof S3FolderNode) {
+        // get files from the folder and convert to S3File
+        files = (await node.s3.listFiles({ bucketName: node.bucket.name, folderPath: node.folder.path })).files.map(
+            file => {
+                return { bucket: node.bucket, ...file }
+            }
+        )
+        // create a folder with the folder name
+        const dir = path.join(saveLocation.fsPath, node.folder.name)
+        if (!fs.existsSync(dir)) {
+            fs.mkdirSync(dir)
+            saveLocation = vscode.Uri.file(dir)
+        }
+    } else {
+        // since we cannot control which nodes are selected and passed here, filter only S3FileNodes
```

Comment:
This is essentially a silent failure. The user selected a node to download but we don't because we can't. But that might not be obvious to everyone.

Reply:
Ok, I can add a log message that describes why the node was ignored.